style: apply /style-guide pass to models/sweeps (#2722)
wandb/docs: models/sweeps/pause-resume-and-cancel-sweeps.mdx

---
description: Pause, resume, and cancel a sweep with the CLI.
title: Manage sweeps
---

This page shows you how to control the lifecycle of an active sweep from the command line. You can stop wasted compute, preserve in-progress experiments, or temporarily halt exploration without losing state.

> Use the [W&B CLI](/models/ref/cli/wandb-sweep) to pause, resume, stop, and cancel a sweep. Each command has a different effect on whether the sweep creates new runs and on runs that are already executing. In each case, provide the sweep ID that W&B generated when you initialized the sweep.

Reviewer: This para seems a bit garbled in the middle.

Author: Reworded the middle sentence: "Each command affects whether the sweep creates new runs and what happens to runs that are already executing."

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Pause a sweep

> Pause a sweep so it temporarily stops creating new runs. Runs that are already executing continue running until completion. Use the [`wandb sweep --pause`](/models/ref/cli/wandb-sweep) command to pause a sweep. Provide the sweep ID that you want to pause.

Reviewer: ```suggestion
Use the [`wandb sweep --pause`](/models/ref/cli/wandb-sweep) command to pause a sweep, so it temporarily stops creating new runs. Runs that are already executing continue to run until completion. Provide the sweep ID.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

```bash
wandb sweep --pause entity/project/sweep_ID
```

## Resume a sweep

> Resume a paused sweep with the [`wandb sweep --resume`](/models/ref/cli/wandb-sweep) command. The sweep starts creating new runs again according to its search strategy. Provide the sweep ID that you want to resume:

Reviewer: ```suggestion
Resume a paused sweep with the [`wandb sweep --resume`](/models/ref/cli/wandb-sweep) command, so it starts creating new runs again according to its search strategy. Provide the sweep ID.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

```bash
wandb sweep --resume entity/project/sweep_ID
```

## Stop a sweep

> Stop a sweep to prevent the creation of new runs while letting executing runs finish gracefully. Use the [`wandb sweep --stop`](/models/ref/cli/wandb-sweep) command:

Reviewer: ```suggestion
Use the [`wandb sweep --stop`](/models/ref/cli/wandb-sweep) command to stop a sweep. A stopped sweep does not create new runs. Runs that are already executing continue to run until completion. Provide the sweep ID.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

```bash
wandb sweep --stop entity/project/sweep_ID
```

## Cancel a sweep

> Cancel a sweep to immediately terminate all active runs and stop creating new runs. This is the only sweep command that forcibly terminates existing runs. Runs terminate abruptly, and the running processes have no chance to run user-defined signal handlers. Use the [`wandb sweep --cancel`](/models/ref/cli/wandb-sweep) command to cancel a sweep. Provide the sweep ID that you want to cancel. For more information about signals and sweep runs, see [Signal handling and sweep runs](/models/sweeps/signal-handling-sweep-runs).

Reviewer: ```suggestion
Use the [`wandb sweep --cancel`](/models/ref/cli/wandb-sweep) command to cancel a sweep, forcibly terminating all active runs and preventing creation of new runs. Runs terminate abruptly, giving the running processes no chance to run user-defined signal handlers. Provide the sweep ID.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

> For a full list of CLI command options, see the [wandb sweep](/models/ref/cli/wandb-sweep) CLI Reference Guide.

Reviewer: ```suggestion
For more information about signals and sweep runs, see [Signal handling and sweep runs](/models/sweeps/signal-handling-sweep-runs). For a full list of CLI command options, see the [wandb sweep](/models/ref/cli/wandb-sweep) CLI Reference Guide.
```

Author: Applied your suggestion (moved the signal-handling reference here).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

<Warning>

> W&B doesn't terminate active [sweeps](/models/sweeps) or agents when you delete a project.

Reviewer: What does this mean? How can you terminate them in this situation?

Author: Clarified: deleting a project doesn't stop active sweeps or their agents, so you must stop or cancel the sweep (`--stop`/`--cancel`) and stop each agent process yourself. Updated the warning to say exactly that.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

</Warning>

## Sweep and run statuses

> A sweep orchestrates multiple runs to explore hyperparameter combinations. To manage your hyperparameter optimization effectively, you must understand how sweep status and run status interact. The following sections describe how the two statuses differ, what happens when you stop an individual run, and which lifecycle command to choose.

Reviewer: ```suggestion
A sweep orchestrates multiple runs to explore hyperparameter combinations. The following sections describe how sweep status and run status interact, so you can manage your hyperparameter optimizations effectively, including how the two statuses differ, what happens when you stop an individual run, and which lifecycle command to choose.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

### Key differences

> - **Sweep status** controls whether the agent creates new runs (Running, Paused, Stopped, Cancelled, Finished, Failed, Crashed).

Reviewer: ```suggestion
- **Sweep status** controls whether the agent creates new runs. Sweep status can be `Running`, `Paused`, `Stopped`, `Cancelled`, `Finished`, `Failed`, or `Crashed`.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

> - **Run status** reflects the execution state of individual runs (Pending, Running, Finished, Failed, Crashed, Killed).

Reviewer: ```suggestion
- **Run status** reflects the execution state of individual runs. Run status can be `Pending`, `Running`, `Finished`, `Failed`, `Crashed`, or `Killed`.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

### Stop an individual run

> When you [stop a run](/models/runs/stop-runs) in a sweep, the sweep agent automatically starts the next run in the sweep. You can skip poorly performing configurations without interrupting the sweep's overall progress.

Reviewer: ```suggestion
When you [stop a run](/models/runs/stop-runs) in a sweep, the sweep agent automatically starts the next run in the sweep. Use this mechanism to skip poorly performing configurations without interrupting the sweep's overall progress.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

> ### Best practices

Reviewer: ```suggestion
### Recommendations
```
May break anchors

Author: Applied. Confirmed there are no inbound links to the `#best-practices` anchor, so nothing breaks.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

The following recommendations help you choose the right lifecycle command for the situation, so you avoid losing useful work or holding onto unwanted compute.

> - Use `--pause` instead of cancel when you want to temporarily halt exploration without losing running experiments.

Reviewer: ```suggestion
- Use `--pause` instead of cancel to temporarily halt exploration without losing running experiments.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
